major/source/MobileClient.cpp

```diff
@@ -0,0 +1,224 @@
+  {
+    const pair<string, string> setData1 = make_pair(createPath(number, PATH_INC_NUM), _number);
+    _netConfAgent->changeData(setData1);
+  }
+
+  void MobileClient::makeCall(const string &number)
+  {
+    setState(_number, BUSY);
+
+    map<string, string> dataForFetch;
+    _netConfAgent->fetchData(createPath(number), dataForFetch);
+    const pair<string, string> setData = make_pair(createPath(_number, PATH_STATE), IDLE);
+
+    for (map<string, string>::const_iterator it = dataForFetch.begin(); it != dataForFetch.end(); it++)
+    {
+      if (it->second == BUSY)
+      {
+        cout << "client is busy now" << endl;
+        const pair<string, string> setData = make_pair(createPath(_number, PATH_STATE), IDLE);
+        _netConfAgent->changeData(setData);
+      }
+      else if (it->second == ACTIVE)
+      {
+        cout << "client is talking" << endl;
+        _netConfAgent->changeData(setData);
+      }
+      else if (it->second == IDLE)
+      {
+        if (dataForFetch.find(createPath(number, PATH_INC_NUM)) == dataForFetch.end())
+        {
+          setIncomigNumber(number);
+          const pair<string, string> setData1 = make_pair(createPath(_number, PATH_INC_NUM), number);
+
+          _netConfAgent->changeData(setData1);
+        }
+        else
+        {
+          cout << "line is busy" << endl;
+          _netConfAgent->changeData(setData);
+        }
+      }
+    }
+  }
+  void MobileClient::answer()
+  {
+    map<string, string> dataForFetch;
+
+    _netConfAgent->fetchData(createPath(_number), dataForFetch);
+    string incomingNumber;
+
+    for (map<string, string>::const_iterator it = dataForFetch.begin(); it != dataForFetch.end(); it++)
+    {
+      if (it->first != createPath(_number, PATH_INC_NUM))
+      {
+        continue;
+      }
+
+      incomingNumber = it->second;
+      cout << incomingNumber << endl;
+      _netConfAgent->deleteItem(createPath(incomingNumber, PATH_INC_NUM));
+      setState(_number, ACTIVE);
+      setIncomigNumber(incomingNumber);
+      setState(incomingNumber, ACTIVE);
+
+      break;
+    }
+  }
+
+  void MobileClient::callEnd()
+  {
+    map<string, string> dataForFetch;
+
+    _netConfAgent->fetchData(createPath(_number), dataForFetch);
+    string incomingNumber = {};
+
+    for (map<string, string>::const_iterator it = dataForFetch.begin(); it != dataForFetch.end(); it++)
+    {
+      if (it->first != createPath(_number, PATH_INC_NUM))
+      {
+        continue;
+      }
+
+      incomingNumber = it->second;
+      _netConfAgent->deleteItem(createPath(_number, PATH_INC_NUM));
+      _netConfAgent->deleteItem(createPath(incomingNumber, PATH_INC_NUM));
+      setState(_number, IDLE);
+      setState(incomingNumber, IDLE);
+      break;
+    }
+  }
+
+  void MobileClient::setState(const string &number, const string &status)
+  {
+    const pair<string, string> setData = make_pair(createPath(number, PATH_STATE), status);
```

Comment:
you can simplify code using auto instead of declaring types)
auto setData = make_pair(createPath(number, PATH_STATE), status);